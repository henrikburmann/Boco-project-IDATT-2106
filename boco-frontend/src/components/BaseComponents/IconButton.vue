<template>
  <!-- Button -->
  <button
    class="flex items-center px-2 py-2 font-medium tracking-wide capitalize text-white transition-colors duration-200 transform rounded-md focus:outline-none focus:ring focus:ring-opacity-80"
    :class="color"
  >
    <div class="w-5 h-5 mx-1">
      <!-- Slot for icon (Default BanIcon) -->
      <slot><BanIcon /></slot>
    </div>
    <span class="mx-1">{{ text }}</span>
  </button>
</template>

<script>
import { BanIcon } from "@heroicons/vue/outline";

/**
 * IconButton component that can be customized with text, color and a slot for passing an icon.
 * Colors:
 * blue (default),
 * red,
 * green
 */
export default {
  name: "IconButton",
  props: {
    text: String,
    buttonColor: String,
  },
  components: {
    BanIcon,
  },
  computed: {
    /**
     * Formats the tailwind css tags based on the color passed from parent
     */
    color() {
      if (this.buttonColor === "red") {
        return "bg-error-medium hover:bg-error-dark focus:ring-error-light";
      }
      if (this.buttonColor === "green") {
        return "bg-success-medium hover:bg-success-dark focus:ring-success-light";
      }
      return "bg-primary-medium hover:bg-primary-dark focus:ring-primary-light";
    },
  },
};
</script>
