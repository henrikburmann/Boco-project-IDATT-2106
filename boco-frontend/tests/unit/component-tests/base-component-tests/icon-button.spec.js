import { mount } from "@vue/test-utils";
import IconButton from "@/components/BaseComponents/IconButton.vue";

describe("IconButtonComponent", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(IconButton, {
      //passing prop to component
      props: {
        text: "hei",
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
