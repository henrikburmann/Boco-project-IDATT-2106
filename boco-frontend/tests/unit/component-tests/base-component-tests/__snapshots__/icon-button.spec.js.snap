// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtonComponent renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <!-- Button -->
  <button
    class="flex items-center px-2 py-2 font-medium tracking-wide capitalize text-white transition-colors duration-200 transform rounded-md focus:outline-none focus:ring focus:ring-opacity-80 bg-primary-medium hover:bg-primary-dark focus:ring-primary-light"
  >
    <div
      class="w-5 h-5 mx-1"
    >
      <!-- Slot for icon (Default BanIcon) -->
      
      <svg
        aria-hidden="true"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.364 18.364A9 9 0 005.636 5.636m12.728 12.728A9 9 0 015.636 5.636m12.728 12.728L5.636 5.636"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      
    </div>
    <span
      class="mx-1"
    >
      hei
    </span>
  </button>
  
</div>
`;
